aptos-move/aptos-release-builder/src/utils.rs

```diff
@@ -106,27 +107,26 @@ pub(crate) fn finish_with_footer(writer: &CodeWriter) -> String {
 pub(crate) fn generate_governance_proposal<F>(
     writer: &CodeWriter,
     is_testnet: bool,
-    next_execution_hash: Vec<u8>,
+    next_execution_hash: Option<HashValue>,
+    is_multi_step: bool,
     deps_names: &[&str],
     body: F,
 ) -> String
 where
     F: FnOnce(&CodeWriter),
 {
-    if next_execution_hash.is_empty() {
-        if is_testnet {
-            generate_testnet_header(writer, deps_names);
-        } else {
-            generate_governance_proposal_header(
-                writer,
-                deps_names,
-                false,
-                "".to_owned().into_bytes(),
-            );
-        }
-    } else {
+    assert!(
+        is_multi_step || next_execution_hash.is_none(),
+        "only multi-step proposals can have a next execution hash"
+    );
+
+    if is_multi_step {
         generate_governance_proposal_header(writer, deps_names, true, next_execution_hash);
-    };
+    } else if is_testnet {
```

Comment:
This looks a bit confusing to me, I think mainly because we pass two params and if `is_multi_step` the hash is none, but we reset it here. But can be a future TODO.

Reply:
Yeah this one is a bit awkward, partially due to clippy forcing me to to change `if {} else { if {} else {} }` into `if {} else if {} else {}`.